### DISCUSSION TITLE
Image Format Conversion

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#5453] [Help]
can someone help on how to use MagickWand API to convert the image from one format to another format

**Comment by snibgo** [comment]
You can read a file using `MagickReadImage()`. Then write it with `MagickWriteImages()`. When writing to a file, the extension will determine the file format. EDIT: Or use a format prefix in the output filename, such a `PNG48:out.png`.

**Reply by programmerarunkumar** [reply]
> When writing to a file, the extension will determine the file format

You mean the file used in MagickWriteImages() should have the extension ?

**Comment by snibgo** [comment]
The filename, yes. For example, `myfile.tiff` if you want TIFF format.

**Reply by programmerarunkumar** [reply]
okay 
Is it necessary to use the file extensions for the file used in MagickReadImage() ?

**Comment by snibgo** [comment]
With `MagickReadImage()`, you are reading a file that already exists, so you must use the name of the file, including the extension if it has one.

**Reply by programmerarunkumar** [reply]
okay Thank you
